src/core/drive/navigator.ts

```diff
@@ -89,8 +89,8 @@ export class Navigator {
     const responseHTML = await fetchResponse.responseHTML
 
     if (responseHTML) {
-      const snapshot = Snapshot.fromHTMLString(responseHTML)
-      this.view.render({ snapshot }, () => {})
+      const snapshot = Snapshot.wrap(responseHTML)
+      this.view.render({ snapshot }, () => window.scroll(0, 0))
```

Comment:
I'm not sure how to manage Intern coverage for the scrolling behavior, but if you have suggestions I'll gladly add the tests.

Reply:
Might want to take a look at the scroll restoration tests: https://github.com/hotwired/turbo/blob/5525c21288edc222de0a05c72538aa7f898abd0c/src/tests/functional/scroll_restoration_tests.ts

Reply:
@sstephenson thank you! I've added some coverage.